lib/Option.cpp

```diff
@@ -294,10 +298,11 @@ bool WorkDim() { return work_dim; }
 bool GlobalOffset() { return global_offset; }
 bool GlobalOffsetPushConstant() { return global_offset_push_constant; }
 bool NonUniformNDRangeSupported() {
-  return (Language() == SourceLanguage::OpenCL_CPP) ||
-         (Language() == SourceLanguage::OpenCL_C_20) ||
-         (Language() == SourceLanguage::OpenCL_C_30) ||
-         cl_arm_non_uniform_work_group_size;
+  return ((Language() == SourceLanguage::OpenCL_CPP) ||
+          (Language() == SourceLanguage::OpenCL_C_20) ||
+          (Language() == SourceLanguage::OpenCL_C_30) ||
+          cl_arm_non_uniform_work_group_size) &&
+         !uniform_workgroup_size;
```

Comment:
It should be an error to have both `cl_arm_non_uniform_work_group_size` and `uniform_workgroup_size` set.

Reply:
PTAL if my change does that as expected

Reply:
Please remove the assert and instead add a check in `ParseOptions()` in lib/Compiler.cpp. You can see examples of error checking for option conflicts there, but the following should be sufficient (after the call to `ParseCommandLineOptions`):
```
if (cl_arm_non_uniform_work_group_size && uniform_workgroup_size) {
  llvm::errs() << "cannot enable Arm non-uniform workgroup extension support and assume uniform workgroup sizes\n";
  return -1;
}
```

Reply:
@alan-baker oops, did I interpret it correctly in the last commit?